lib/device-manager.js

```diff
@@ -6,29 +7,174 @@
  * Keeps track of registered devices.
  */
 
+var async = require('async');
+var execSync = require('child_process').execSync;
 var path = require('path');
 var config = require('./config');
 var Device = require('./device');
 
-function DeviceManager () {
+/**
+ *
+ * @param oslevel String of the form '9.1'
+ * @param line String of the form 'iPhone 4s (05686243-BA24-44CE-94D9-D6823BF96E46) (Shutdown) {(unavailable, ...)}'
+ * @returns {{tag: string}}
+ */
+
+/**
+ *
+ * @param oslevel String
+ * @param line String to parse
+ * @returns Device-like JSON object
+ */
+function createDevice(oslevel, line) {
+	if (line.indexOf('unavailable') !== -1) throw new Error('createDevice-fed an unavailable device:' + oslevel + ':' +  line);
+	var openparenIndex = line.indexOf('(');
+	var closeParenIndex = line.indexOf(')');
+	if (openparenIndex === -1) throw new Error("createDevice: could not find open paren in: " + oslevel + ':' + line);
+	if (closeParenIndex === -1) throw new Error("createDevice: could not find close paren in: " + oslevel + ':' + line);
+	var identifier = line.slice(openparenIndex + 1, closeParenIndex);
+	if (identifier.length < 10) throw new Error('createDevice: could not find valid identifier in:' + oslevel + ':' + line);
+	var name = line.slice(0, openparenIndex).trim();
+	var osVersion = oslevel.split(' ')[1]; // oslevel is of form 'iOS 9.1'
+	return {
+		tag: 'ios-simulator,OS=' + osVersion + ',name=' + name,
+		platform: 'ios',
+        identifier: identifier,
+		name: 'ios-simulator ' + name + ' (' + osVersion + ')',
+		OS: oslevel,
+		simulator: true,
+		destination: 'OS=' + osVersion + ',name=' + name
+	};
+}
+
+function isHeader(line) {
+	var snippet =  line.slice(0,2);
+	return (snippet === '--' || snippet === '==');
+}
+
+/**
+ * A line oriented Stream over Simctl output from running $xcrun simctl list
+ * @param array {Array.<String>}
+ * @returns {{next: Function, advanceTillMatch: Function, hasMore: Function, nextValue: Function, nextValueAsOSLevel: Function, peekHasNextLineEntry: Function, peekIsHeader: Function, peekIsDoubleEquals: Function, peek: Function}}
+ */
+function makeSimctlStream(array){
```

Comment:
I'd also avoid using `array` as an arg name. Very close to `Array`, which I'm assuming should be treated like a reserved word.